### ISSUE TITLE
Do custom classes for finetuning have to be appended to the class names/indices of the pre-trained model in the dataset configuration?

### ISSUE THREAD

**Issue by vankeer** [ultralytics/ultralytics#833]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, my apologies if this question is quite basic, but I'm new to YOLO and haven't found a clear answer anywhere yet.

When finetuning a pre-trained YOLOv8 model, how should the class indices be defined (both in the label txt files and the dataset yaml)? Should the class names of the pre-trained model be taken into account?

The reason I'm asking is because I'm not sure if re-using class indices 0 ... 79 would confuse the model due to conflicts with the 80 classes defined in the COCO128 dataset that was used to train the standard models?

If such confusion is possible, do we have to use label indices bigger than 79 and define the classes like this:

```
names:
- person
- bicycle
- car
- ... # all class names from the pre-trained model
- # custom names here
nc: # number of classes in the pre-trained model + custom classes for finetuning
train: ./train/images
val: ./val/images
```

Thanks

**Comment by knoppmyth** [comment]
When you're transfer learning you don't appending the class names. With respect to the label text file, it doesn't have the name but a number for the class.
`0 0.09439124487004104 0.59521484375 0.15595075239398085 0.1220703125
0 0.2024623803009576 0.55322265625 0.14227086183310533 0.0732421875
0 0.2708618331053352 0.61328125 0.2708618331053352 0.125
0 0.7291381668946648 0.4423828125 0.09575923392612859 0.06640625`

The first 0 there is the class.

For the dataset YAML file:
path: /usr/src/datasets/Hand/
train:
    - V1/images/train
    - V2/images/train
val: V1/images/val
test:  
nc: 1
names: ['Hand']

The "0" in the label would correspond to the "Hand" in the dataset YAML.

**Comment by pderrenger** [comment]
@knoppmyth thank you for sharing your insights! Indeed, when performing transfer learning with YOLOv8, there's no need to append the class names when finetuning. The label text files represent the classes by their corresponding numerical index, and in the dataset YAML file, the "names" list specifies the class names. Consequently, in your dataset YAML file example, the "0" in the label corresponds to the "Hand" class defined in the YAML file. This allows for seamless transfer learning without the need to redefine or append class names or indices from the pre-trained model. If you have any more questions or need further assistance, feel free to ask!
